rafthttp/status.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rafthttp
+
+import (
+	"fmt"
+	"log"
+	"sync"
+
+	"github.com/coreos/etcd/pkg/types"
+)
+
+var (
+	statusMapMu sync.Mutex
+	statusMap   = make(map[string]connStatus)
+)
+
+type connStatus struct {
+	active bool
+	errm   map[string]bool
+}
+
+func report(local, to types.ID, src string, active bool, err *opError) {
+	statusMapMu.Lock()
+	defer statusMapMu.Unlock()
+	name := fmt.Sprintf("%s %s %s", local, to, src)
```

Comment:
this is terrible. the key of a map can be a struct. do not print it out as the key